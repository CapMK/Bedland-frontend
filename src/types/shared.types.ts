export type Getters<T extends string, K> = {
  [k in T]: K;
};

export type Building = {
  id: string;
  name: string;
  address: string;
  managerId: string;
  floors: string;
  createDate: string;
  updateDate: string;
  version: number;
};

export type Flat = {
  id: string;
  version: number;
  createDate: string;
  updateDate: string;
  buildingId: string;
  number: string;
  floor: string;
  shapePath: string;
};

export type FlatWithAddress = Flat & { address: string; city: string };

export type ResidentsFlats = {
  flatId: string;
  flatNumber: string;
  flatAddress: string;
  flatCreateDate: string;
};

export type BuildingWithCity = Building & { city: string };

export type ResidentsFlatsWithCity = ResidentsFlats & { city: string };
