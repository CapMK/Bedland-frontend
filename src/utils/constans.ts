export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
