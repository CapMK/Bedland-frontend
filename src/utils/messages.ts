const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
