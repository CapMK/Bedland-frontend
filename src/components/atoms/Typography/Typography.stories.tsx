import type { Meta, StoryObj } from '@storybook/react';
import { colorNames, typographyNames } from 'theme/theme.types';
import ThemeProvider from 'theme/ThemeContext';
import Typography from './Typography';

const meta = {
  title: 'Atoms/Typography',
  component: Typography,
  tags: ['autodocs'],
  argTypes: {
    variant: {
      options: typographyNames,
      control: 'radio',
      description: 'Typography variant',
      defaultValue: { summary: 'paragraph' },
      table: {
        type: {
          summary: `Variants: ${typographyNames.toString().replaceAll(',', ' | ')}`,
        },
        defaultValue: { summary: 'paragraph' },
      },
    },
    children: {
      name: 'label',
      description: 'Text to display',
    },
    style: {
      table: {
        disable: true,
      },
    },
    color: {
      options: colorNames,
      control: 'select',
      description: 'Color of the text',
      defaultValue: 'default',
    },
  },
  decorators: [(Story) => <ThemeProvider>{Story()}</ThemeProvider>],
  parameters: {
    backgrounds: { default: 'light' },
  },
} satisfies Meta<typeof Typography>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Header2: Story = {
  args: {
    variant: 'header2',
    color: 'text',
    children: 'header2',
  },
};

export const Paragraph: Story = {
  args: {
    variant: 'paragraph',
    color: 'text',
    children:
      'There was nothing so very remarkable in that; nor did Alice think it so very much out of the way to hear the Rabbit say to itself, - Oh dear! Oh dear! I shall be late!- (when she thought it over afterwards, it occurred to her that she ought to have wondered at this, but at the time it all seemed quite natural); but when the Rabbit actually took a watch out of its waistcoat-pocket, and looked at it, and then hurried on, Alice started to her feet, for it flashed across her mind that she had never before seen a rabbit with either a waistcoat-pocket, or a watch to take out of it, and burning with curiosity, she ran across the field after it, and fortunately was just in time to see it pop down a large rabbit-hole under the hedge. ',
  },
};

export const Captions: Story = {
  args: {
    variant: 'captions',
    color: 'text',
    children: 'Caption',
  },
};

export const SidebarItem: Story = {
  args: {
    variant: 'header4',
    color: 'text',
    children: 'header 4',
  },
};
