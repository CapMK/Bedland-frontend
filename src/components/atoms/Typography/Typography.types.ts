import { CSSProperties } from 'react';
import { ColorNames, TypographyName } from 'theme/theme.types';

export type TypographyProps = {
  variant?: TypographyName;
  color?: ColorNames;
  style?: CSSProperties;
};
