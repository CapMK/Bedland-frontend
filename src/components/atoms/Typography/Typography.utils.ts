import { TypographyVariants } from 'theme/theme.types';

const variantToElement: TypographyVariants = {
  header1: 'h1',
  header2: 'h2',
  subtitles: 'h2',
  header3: 'h3',
  header4: 'h4',
  header5: 'h5',
  subHeader: 'h5',
  paragraph: 'p',
  dataInputsAndTooltips: 'p',
  unerlined: 'p',
  captions: 'p',
} as const;

export default variantToElement;
