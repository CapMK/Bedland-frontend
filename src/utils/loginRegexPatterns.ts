import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
