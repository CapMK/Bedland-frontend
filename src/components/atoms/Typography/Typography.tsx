import { CSSProperties, PropsWithChildren, useContext } from 'react';
import { ThemeContext } from 'theme/ThemeContext';
import { TypographyProps } from './Typography.types';
import variantToElement from './Typography.utils';

const Typography = ({ variant, style, color, children }: PropsWithChildren<TypographyProps>) => {
  const { palette, typography } = useContext(ThemeContext);
  const Element = variantToElement[variant ?? 'paragraph'];

  const colorStyles: CSSProperties = {
    color: color ? palette[color] : 'none',
  };

  return (
    <Element
      style={{
        ...typography[variant ?? 'paragraph'],
        ...colorStyles,
        ...style,
      }}
    >
      {children}
    </Element>
  );
};

export default Typography;
