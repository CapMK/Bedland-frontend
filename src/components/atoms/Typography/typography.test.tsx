import { cleanup } from '@testing-library/react';
import { PropsWithChildren } from 'react';
import { renderWithProviders } from 'utils/test.utils';
import Typography from './Typography';
import { TypographyProps } from './Typography.types';

describe('Typography', () => {
  afterEach(cleanup);
  const item = ({ variant, color, children }: PropsWithChildren<TypographyProps>) =>
    renderWithProviders(
      <Typography variant={variant} color={color}>
        {children}
      </Typography>,
      {},
    );

  it('should renders correctly typography - paragraph', () => {
    expect(
      item({ variant: 'paragraph', color: 'blueDark', children: 'Paragraph' }),
    ).toMatchSnapshot();
  });

  it('should renders correctly typography - header4', () => {
    expect(item({ variant: 'header4', color: 'blueDark', children: 'header4' })).toMatchSnapshot();
  });
});
