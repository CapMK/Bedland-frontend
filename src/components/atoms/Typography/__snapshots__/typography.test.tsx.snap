// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Typography > should renders correctly typography - header4 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h4
        style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
      >
        header4
      </h4>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <h4
      style="font-family: Source Sans Pro, sans-serif; font-size: 1rem; font-weight: 600; line-height: 1.4; letter-spacing: 1px; color: rgb(69, 83, 221);"
    >
      header4
    </h4>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Typography > should renders correctly typography - paragraph 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(69, 83, 221);"
      >
        Paragraph
      </p>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <p
      style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(69, 83, 221);"
    >
      Paragraph
    </p>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
